pkg/controller/compliancescan/compliancescan_controller.go

```diff
@@ -649,6 +650,26 @@ func (r *ReconcileComplianceScan) phaseAggregatingHandler(h scanTypeHandler, log
 		instance.Status.ErrorMessage = err.Error()
 	}
 
+	// count the number of checks that were run
+	checkCount, err := r.fetchResultCounts(instance, logger)
+	if err != nil {
+		logger.Error(err, "Cannot fetch the number of checks")
```

Comment:
Similar comment here as below. But we could reduce the duplication too and log a single error message either here or in `fetchResultCounts`.

Reply:
>Similar comment here as below. But we could reduce the duplication too and log a single error message either here or in fetchResultCounts.

that's sounds good, let me fix that

`Also - is my understanding correct that this is effective a fatal error since we're not putting the request back in the queue?` I think it will still requeue?  an error will be logged here, since we are not waiting on anything, it should return error, and retry later

Reply:
Ah - yes, you're right.

https://kubebyexample.com/learning-paths/operator-framework/operator-sdk-go/controller-reconcile-function

Returning the error will requeue the request.